includes/templates/wcdp_step_1.php

```diff
@@ -1,126 +1,123 @@
                 <div></div>
                 <div></div>
             </div>
-        <?php elseif ($value['style'] == '4') : ?>
-            <button class="button wcdp-button wcdp-right" type="submit">
-                <?php esc_html_e('Donate', 'wc-donation-platform'); ?>&nbsp;<div class="wcdp-arrow wcdp-right-arrow">
+		<?php elseif ( $value[ 'style' ] == '4' ) : ?>
+			<button class="button wcdp-button wcdp-right" type="submit">
+                <?php esc_html_e( 'Donate', 'wc-donation-platform' ); ?>&nbsp;<div class="wcdp-arrow wcdp-right-arrow">
                     &raquo;
                 </div>
             </button>
-        <?php endif; ?>
-        <div class="wcdp-divider"></div>
+		<?php endif; ?>
+		<div class="wcdp-divider"></div>
 
-        <?php //WooCommerce Add to Cart & Quantity (invisible) ?>
-        <input style="display:none !important;" type="number" name="quantity" class="quantity qty" value="1">
-        <?php
-        /** @var bool $is_internal */
-        if ($value['style'] == 4 && $is_internal) {
-            echo '<input class="wcdp-express-amount" style="display:none !important;" type="number" step="any" name="attribute_wcdp_donation_amount" value="1">';
-            do_action('wcdp_express_checkout_heading');
-            do_action('woocommerce_after_add_to_cart_quantity');
-        }
-        ?>
-        <button style="display:none !important;" type="submit" name="add-to-cart"
-                value="<?php echo esc_attr($product->get_id()); ?>"
-                class="single_add_to_cart_button button alt"></button>
-        <?php do_action('woocommerce_after_add_to_cart_button'); ?>
+		<?php //WooCommerce Add to Cart & Quantity (invisible) ?>
+		<input style="display:none !important;" type="number" name="quantity" class="quantity qty" value="1">
+		<?php
+		/** @var bool $is_internal */
+		if ( $value[ 'style' ] == 4 && $is_internal ) {
+			echo '<input class="wcdp-express-amount" style="display:none !important;" type="number" step="any" name="attribute_wcdp_donation_amount" value="1">';
+			do_action( 'wcdp_express_checkout_heading' );
+			do_action( 'woocommerce_after_add_to_cart_quantity' );
+		}
+		?>
+		<button style="display:none !important;" type="submit" name="add-to-cart" value="<?php echo esc_attr( $product->get_id() ); ?>" class="single_add_to_cart_button button alt"></button>
+		<?php do_action( 'woocommerce_after_add_to_cart_button' ); ?>
     </form>
 
 <?php
-if ($value['style'] == 4 && $is_internal) {
-    //Add a input field to communicate correct donation amount to Stripe & PayPal Express donation buttons
-    do_action('wcdp_express_checkout_amount_variation');
-
-    do_action('woocommerce_after_add_to_cart_form');
+if ( $value[ 'style' ] == 4 && $is_internal ) {
+	//Add a input field to communicate correct donation amount to Stripe & PayPal Express donation buttons
+	do_action( 'wcdp_express_checkout_amount_variation' );
 }
+
+do_action( 'woocommerce_after_add_to_cart_form' );
```

Comment:
@jonas-hoebenreich I changed the location of this line `do_action( 'woocommerce_after_add_to_cart_form' );` because, in the previous code this line available in if the condition and hook do not work.

Reply:
yes, in the other styles this hook is called in step 2 in order to make the plugin compatible with Stripe Apple Pay & Google Pay: https://github.com/jonas-hoebenreich/wc-donation-platform/blob/main/includes/templates/wcdp_step_2.php#L15